src/core/ddsc/src/dds_reader.c

```diff
@@ -490,7 +490,11 @@ static bool dds_reader_support_shm(const struct ddsi_config* cfg, const dds_qos_
   if(!(qos->durability.kind == DDS_DURABILITY_VOLATILE ||
     qos->durability.kind == DDS_DURABILITY_TRANSIENT_LOCAL)) {
     return false;
-  }  
+  }
+
+  if(qos->present & QP_CYCLONE_IGNORELOCAL && qos->ignorelocal.value != DDS_IGNORELOCAL_NONE) {
```

Comment:
Shouldn't this also be checked for the writing side?

Reply:
I do not think so. From my understanding ignorelocal implies: the reader cannot use shared memory if it is set to ignore local publishers (as they might send something with iceoryx which it would receive).

I am not sure what the meaning would be on writer side: ignore any local readers? Then we should indeed check it on the writer side as well (and if it iset to ignore local readers, it will not use iceoryx). But I think it is sufficient if the reader basically tells the writer that it wants to be ignored.

Ideally iceoryx could deal with this (by setting a flag on its subscriber), but it cannot without creating further issues.

Reply:
Add on writer side as well.